### PR TITLE
Structure and clarify `limitations.md`

### LINKED ISSUE #345: Clarify when/whether limitations are expected to be resolved
While currently `limitations.md` describes **what** kind of limitations exists, it doesn't give any indication **when** and **whether** they are expected to be resolved. (Especially in regards to the GA release.) We should clarify this.

### PR DESCRIPTION
**Pull request by janpio**
([Rendered View](https://github.com/prisma/prisma2/blob/janpio-limitations/docs/limitations.md))

This PR splits the existing list of limitations into categories (Functionality, Performance, Stability) and adds more tracking issues and information. It also puts the "Out of Scope" items into their own separate list so it is clearer what is and what is not planned.

The information in here is not nearly exhaustive and we should spend more effort in the future to fix this. (@pantharshit00 for example will add a few more tracking issues and link them here)

closes #345

(To be merged via Squash+Merge)

Review Requests:
@divyenduz for support, @nikolasburk for content/marketing, @schickling and @sorenbs for product

### CODE REVIEW

File: docs/limitations.md [hunk 1]
```diff
@@ -1,12 +1,30 @@
 # Current limitations
 
-1. Does not work on Windows. Tested on Mac and Linux only. Tracking issue: https://github.com/prisma/prisma2/issues/4
-2. Processes 1 request at a time, no parallelism during the early preview period. This will be fixed before GA. 
-3. Some edge cases for complex nested mutations don't work properly. 
-4. Limited auto-complete in Typescript projects due to a compiler bug. We raised a PR to fix that [here](https://github.com/microsoft/TypeScript/pull/32100). 
-5. No realtime API/subscriptions. In the future, we will have Prisma events engine (after the GA) that would pave way for such a feature, there is no ETA for this yet. 
-6. Models must have an `@id` attribute and it must take one of these forms:
+During the preview phase of Prisma 2 there are missing features, limited performance and stability issues you should be aware of:
```
Comment (nikolasburk) [inline docs/limitations.md]: Nitpick: We've typically uppercased "Preview". Rationale: "General Availability" is typically also uppercased. It's basically a _name_ for a certain phase of the software lifecycle.

File: docs/limitations.md [hunk 2]
```diff
@@ -1,12 +1,30 @@
...
+
+## Functionality limitations
+
+- `Embed` types are not implemented yet ([tracking issue](https://github.com/prisma/lift/issues/43))
```
Comment (nikolasburk) [inline docs/limitations.md]: I'd suggest to only use inline code formatting when this is actually referencing some programmatic concept, so here maybe either lowercase `embed` or remove the inline code formatting.

File: docs/limitations.md [hunk 3]
```diff
@@ -1,12 +1,30 @@
...
+- Many other types are not fully implemented yet
+- Models must have an `@id` attribute and it must take one of these forms:
     - `Int @id`
     - `String @id @default(uuid())`
-    - `String @id @default(cuid())`
-7. When [introspecting](./introspection.md) a database, Prisma only recognizes many-to-many relations that follow the Prisma conventions for [relation tables](https://github.com/prisma/prisma2/blob/master/docs/relations.md#mn).
+    - `String @id @default(cuid())`  
+- When [introspecting](./introspection.md) a database, Prisma for now only recognizes many-to-many relations that follow the Prisma conventions for [relation tables](https://github.com/prisma/prisma2/blob/master/docs/relations.md#mn).
+- Some edge cases for complex nested mutations don't work properly yet.
+- Non-interactive TTYs (like Git Bash on Windows) are currently not supported by Prisma2 CLI ([tracking issues](https://github.com/prisma/prisma2/issues/554))
```
Comment (nikolasburk) [inline docs/limitations.md]: I had to look up what a TTY is 🙈

File: docs/limitations.md [hunk 4]
```diff
@@ -1,12 +1,30 @@
...
+
+## Stability limitations
+
+- Limited auto-complete in Typescript projects due to a compiler bug ([tracking issue](https://github.com/microsoft/TypeScript/issues/30507) over at `TypeScript`, and the [PR we raised that fixes that](https://github.com/microsoft/TypeScript/pull/32100) and waits to get merged)
```
Comment (nikolasburk) [inline docs/limitations.md]: casing: TypeScript

**Review by nikolasburk — CHANGES_REQUESTED** [review]
Looks great, I've made a few minor suggestions. Also, to be consistent I'd suggest to use proper punctuation on all bullet points (i.e. end each sentence with a full stop 🙂).

File: docs/limitations.md (hunk 4 above)
Comment (janpio) [reply docs/limitations.md]: You can directly suggest changes here in the review that I can directly accept using the button left next to `"`:

```suggestion
- Limited auto-complete in TypeScript projects due to a compiler bug ([tracking issue](https://github.com/microsoft/TypeScript/issues/30507) over at `TypeScript`, and the [PR we raised that fixes that](https://github.com/microsoft/TypeScript/pull/32100) and waits to get merged)
```

File: docs/limitations.md (hunk 1 above)
Comment (janpio) [reply docs/limitations.md]: I should remove the "phase" then, right?

File: docs/limitations.md (hunk 2 above)
Comment (janpio) [reply docs/limitations.md]: Isn't this a type from our schema? https://github.com/prisma/specs/tree/master/schema#embed-block
But yeah, that is still the "concept" referred to here, not the `embed` actually used in the schema, right?

Comment (janpio) [reply docs/limitations.md]: Workaround (which also matches the issue):
```suggestion
- Embedded types are not implemented yet ([tracking issue](https://github.com/prisma/lift/issues/43))
```
Correct?

Comment (nikolasburk) [reply docs/limitations.md]: > Isn't this a type from our schema? https://github.com/prisma/specs/tree/master/schema#embed-block

Absolutely! My point was to spell it like this type if you're using inline code (which means it should be lowercased). Sorry if that wasn't clear!

File: docs/limitations.md (hunk 3 above)
Comment (janpio) [reply docs/limitations.md]: How can we fix that? Also not happy with using that, but "non interactive tty" is a thing people say.

File: docs/limitations.md (hunk 1 above)
Comment (nikolasburk) [reply docs/limitations.md]: Yup I think that sounds a bit better. But I think it's generally fine to speak of the "Preview period" or "Preview phase".

File: docs/limitations.md (hunk 2 above)
Comment (janpio) [reply docs/limitations.md]: I wasn't aware of that before actually visiting that link ;)

What do you think of the suggestion?

File: docs/limitations.md (hunk 1 above)
Comment (janpio) [reply docs/limitations.md]: "Preview period" is best, changed it.

File: docs/limitations.md (hunk 2 above)
Comment (nikolasburk) [reply docs/limitations.md]: Looks good 👍

File: docs/limitations.md (hunk 3 above)
Comment (nikolasburk) [reply docs/limitations.md]: maybe just call it "shells" instead? I'm still not sure what the exact meaning of TTY is here but is there a simpler term for this?

Comment (janpio) [reply docs/limitations.md]:
```suggestion
- Non-interactive shells/TTYs (like Git Bash on Windows) are currently not supported by Prisma2 CLI ([tracking issues](https://github.com/prisma/prisma2/issues/554))
```

Comment (janpio) [reply docs/limitations.md]: Ping @pimeys, I assume you know a bit about that and might help us out if my suggestion is a valid one.

File: docs/limitations.md [hunk 5]
```diff
@@ -1,12 +1,30 @@
...
+- Embedded types are not implemented yet ([tracking issue](https://github.com/prisma/lift/issues/43))
+- Many other types are not fully implemented yet
+- Models must have an `@id` attribute and it must take one of these forms:
     - `Int @id`
     - `String @id @default(uuid())`
-    - `String @id @default(cuid())`
-7. When [introspecting](./introspection.md) a database, Prisma only recognizes many-to-many relations that follow the Prisma conventions for [relation tables](https://github.com/prisma/prisma2/blob/master/docs/relations.md#mn).
+    - `String @id @default(cuid())`  
+- When [introspecting](./introspection.md) a database, Prisma for now only recognizes many-to-many relations that follow the Prisma conventions for [relation tables](https://github.com/prisma/prisma2/blob/master/docs/relations.md#mn).
+- Some edge cases for complex nested mutations don't work properly yet.
```
Comment (nikolasburk) [inline docs/limitations.md]: this should be "nested write" not "nested mutation"

**Review by divyenduz — APPROVED** [review]
LGTM - as long as we commit to everything that is not in "Out of scope functionality"

Without scoping: LGTM

File: docs/limitations.md (hunk 3 above)
Comment (pimeys) [reply docs/limitations.md]: TTY = Terminal Type.

Comment (pimeys) [reply docs/limitations.md]: It's a very common term when talking about computers.

File: docs/limitations.md (hunk 5 above)
Comment (janpio) [reply docs/limitations.md]: Ok, that was just copy/paste from the original thing. Was something implemented in the meantime that changes this?

**Comment by janpio** [comment]
> as long as we commit to everything that is not in "Out of scope functionality"

What does that mean?

File: docs/limitations.md (hunk 3 above)
Comment (janpio) [reply docs/limitations.md]: Thanks.
Is it valid to do "shells/TTYs"?

Comment (pimeys) [reply docs/limitations.md]: Shell is a program such as bash or zsh. Terminal is a program such as the Windows terminal application, xterm, linux console et.al. emulating the original teletype. So shell is wrong here, and you can use an internactive terminal on Windows and with that, bash.

Comment (janpio) [reply docs/limitations.md]: @nikolasburk Your call on this then - I would just keep "TTY" and maybe provide an appropriate link that maybe explains what "non interactive tty" actually is or not.

Comment (pimeys) [reply docs/limitations.md]: (or use the word terminal)

Comment (janpio) [reply docs/limitations.md]:
```suggestion
- Non-interactive terminals (like Git Bash on Windows) are currently not supported by Prisma2 CLI ([tracking issues](https://github.com/prisma/prisma2/issues/554))
```

Comment (janpio) [reply docs/limitations.md]: That definitely works for me. Thanks @pimeys

File: docs/limitations.md (hunk 5 above)
Comment (janpio) [reply docs/limitations.md]: This is now tracked in https://github.com/prisma/prisma2/issues/602

**Comment by janpio** [comment]
Got OK from @schickling on the Out of Scope content in Slack, so merged this.

**Pull request merged**